lib/ip_list_write.go

```diff
@@ -161,54 +160,105 @@ func IPListWriteFromIPRangeStr(rStr string) error {
 
 // IPListWriteFromReader returns a list of IPs after reading from a reader; the
 // reader should have IPs per-line.
-func IPListWriteFromReader(r io.Reader) {
+func IPListWriteAllFromReader(r io.Reader) {
+	IPListWriteFromReader(r, true, true, true)
+}
+
+// IPListWriteFromReader returns a list of IPs after reading from a reader
+// from selected sources; the reader should have IPs per-line.
+func IPListWriteFromReader(
+	r io.Reader,
+	ip bool,
+	iprange bool,
+	cidr bool) {
 	scanner := bufio.NewScanner(r)
 	for scanner.Scan() {
 		ipStr := strings.TrimSpace(scanner.Text())
 		if ipStr == "" {
 			break
 		}
-
-		if err := IPListWriteFromIPRangeStr(ipStr); err == nil {
-			continue
-		}
-
-		if StrIsIPStr(ipStr) {
-			fmt.Println(ipStr)
-			continue
+		if iprange {
+			if err := IPListWriteFromIPRangeStr(ipStr); err == nil {
+				continue
+			}
 		}
-
-		if StrIsCIDRStr(ipStr) {
-			if err := IPListWriteFromCIDR(ipStr); err == nil {
+		if ip {
+			if StrIsIPStr(ipStr) {
+				fmt.Println(ipStr)
 				continue
 			}
 		}
+		if cidr {
+			if StrIsCIDRStr(ipStr) {
+				if err := IPListWriteFromCIDR(ipStr); err == nil {
+					continue
+				}
+			}
+		}
 
 		// simply ignore anything else.
 	}
 }
 
 // IPListWriteFromStdin returns a list of IPs from a stdin; the IPs should be 1
 // per line.
-func IPListWriteFromStdin() {
-	IPListWriteFromReader(os.Stdin)
+func IPListWriteAllFromStdin() {
+	IPListWriteAllFromReader(os.Stdin)
+}
+
+// IPListWriteFromStdin returns a list of IPs from a stdin from selected
+//sources; the IPs should be 1 per line.
+func IPListWriteFromStdin(
+	ip bool,
+	iprange bool,
+	cidr bool) {
+	IPListWriteFromReader(os.Stdin, ip, iprange, cidr)
 }
 
 // IPListWriteFromFile returns a list of IPs found in a file.
-func IPListWriteFromFile(pathToFile string) error {
+func IPListWriteAllFromFile(pathToFile string) error {
 	f, err := os.Open(pathToFile)
 	if err != nil {
 		return err
 	}
 
-	IPListWriteFromReader(f)
+	IPListWriteAllFromReader(f)
+	return nil
+}
+
+// IPListWriteFromSrcFile returns a list of IPs from selected sources found
+// in a file.
+func IPListWriteFromFile(pathToFile string,
+	ip bool,
+	iprange bool,
+	cidr bool) error {
+	f, err := os.Open(pathToFile)
+	if err != nil {
+		return err
+	}
+	IPListWriteFromReader(f, ip, iprange, cidr)
+	return nil
+}
+
+// IPListWriteFromFiles returns a list of IPs found in a list of files from
+// select sources.
+func IPListWriteFromFiles(paths []string,
+	ip bool,
+	iprange bool,
+	cidr bool) error {
+	for _, p := range paths {
+		if err := IPListWriteFromFile(p, ip, iprange, cidr); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
-// IPListWriteFromFiles returns a list of IPs found in a list of files.
-func IPListWriteFromFiles(paths []string) error {
+// IPListWriteFromFiles returns a list of IPs found in a list of files from
+// all sources.
+func IPListWriteAllFromFiles(paths []string) error {
```

Comment:
As above, the implementation of this should be just calling `IPListWriteFromFiles` with all flags turned on.

Reply:
Also, to follow the pattern of the rest of the code structure, the "All" version of a function should be defined above the non-All version. So this should go above `IPListWriteFromFiles`.